packages/back-end/src/models/QueryModel.ts

```diff
@@ -156,10 +160,11 @@ export async function createNewQuery({
     language,
     organization,
     query,
-    startedAt: new Date(),
-    status: result ? "succeeded" : error ? "failed" : "running",
+    startedAt: result || error ? new Date() : undefined,
+    status: result ? "succeeded" : error ? "failed" : "queued",
```

Comment:
For queries that are started immediately, it's better to create them with the right status from the start instead of creating and then updating immediately after.

Reply:
I also only create a new query from a cached query when I'm done checking the cached query status, instead of creating a new query and then copying forward information from the cached query. IDK if there's risk there but it seemed like the right approach.